src/parcsr_ls/par_mgr_setup.c

```diff
@@ -177,6 +177,16 @@ hypre_MGRSetup( void               *mgr_vdata,
    hypre_MPI_Comm_size(comm, &num_procs);
    hypre_MPI_Comm_rank(comm, &my_id);
 
+   /* Reset print_level codes. This is useful for printing
```

Comment:
@victorapm why do these codes need to be reset? Do they change their state?

Reply:
This is useful for the following scenario:
1) User calls `HYPRE_MGRSetup`
2) Current LS data is printed
3) Current LS is solved
4) User *does not* call `HYPRE_MGRDestroy`
5) New LS is assembled and passed in to `HYPRE_MGRSetup`
6) Now we have a new LS to be dumped to file, the reserved bits will signal that we need to print the new data

Reply:
> Do they change their state?

- `((print_level & HYPRE_MGR_PRINT_RESERVED_B) == HYPRE_MGR_PRINT_RESERVED_B)` checks if the result of a bitwise AND operation between print_level and `HYPRE_MGR_PRINT_RESERVED_B` is equal to `HYPRE_MGR_PRINT_RESERVED_B`. In other words, it checks if certain bits in print_level are set to the same values as in `HYPRE_MGR_PRINT_RESERVED_B`.
- If the condition above is true, then `HYPRE_MGR_PRINT_INFO_MATRIX` is bitwise OR'ed with `print_level`, i.e., certain bits in print_level will be set to 1 if they are active in `HYPRE_MGR_PRINT_INFO_MATRIX, effectively turning on some print flags.
- if the condition is false, 0 is bitwise OR'ed with `print_level`, effectively leaving `print_level` unchanged.

Same reasoning for the other checks